File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Examinations/ExaminationsForm.ts
import { ServiceLookupEditor, StringEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ExaminationsForm {
    AppointmentId: ServiceLookupEditor;
    DoctorId: ServiceLookupEditor;
    PatientId: ServiceLookupEditor;
    Complaint: StringEditor;
    ExaminationDate: DateEditor;
    DoctorNote: StringEditor;
}

export class ExaminationsForm extends PrefixedContext {
    static readonly formKey = 'Examinations.Examinations';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ExaminationsForm.init)  {
            ExaminationsForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = StringEditor;
            var w2 = DateEditor;

            initFormType(ExaminationsForm, [
                'AppointmentId', w0,
                'DoctorId', w0,
                'PatientId', w0,
                'Complaint', w1,
                'ExaminationDate', w2,
                'DoctorNote', w1
            ]);
        }
    }
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Examinations/Examinations/ExaminationsForm.cs
using Serenity.ComponentModel;
using System;

namespace MuayeneYonetimPortali.Examinations.Forms;

[FormScript("Examinations.Examinations")]
[BasedOnRow(typeof(ExaminationsRow), CheckNames = true)]
public class ExaminationsForm
{
    public int AppointmentId { get; set; }
    public int DoctorId { get; set; }
    public int PatientId { get; set; }
    public string Complaint { get; set; }
    public DateTime ExaminationDate { get; set; }
    public string DoctorNote { get; set; }
}